lib/rails_autoscale_agent/worker_adapters/delayed_job.rb

```diff
@@ -89,8 +89,12 @@ def track_long_running_jobs?
       end
 
       def select_rows(sql)
+        previous_level = ::ActiveRecord::Base.logger.level
+        ::ActiveRecord::Base.logger.level = ::Logger::WARN if previous_level < ::Logger::WARN
```

Comment:
Thanks @willkoehler! I hear you on the noisiness of these queries.

My main concern here is thread-safety. Could this affect other threads sharing the logger?

I assume this is why Rails has the [`LoggerThreadSafeLevel` module](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/logger_thread_safe_level.rb#L41-L46), which is used by the [`silence` method](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/logger_silence.rb#L17-L19).

Perhaps we could use that instead?

We'll also need to check for the existence of the method since the logger could be any kind of logger.

Reply:
Just saw you commented on this 23 days ago and I missed the notification 🤦‍♂️ That's a good point. I'll circle back to this Friday (on my flex day) and take a closer look.

Reply:
Thanks for pointing out `LoggerSilence` I wasn't aware of that module, but `silence` is clearly the right tool for this job and I appreciate that it's also thread-safe. I updated the PR to use that module.

(BTW I also updated my other applications to use `LoggerSilence` instead of the `::ActiveRecord::Base.logger.level` hack. So much cleaner 🙌)